src/util/base.h

```diff
@@ -0,0 +1,101 @@
+ * Redistribution and use in source and binary forms, with or without modification, are       //
+ * permitted provided that the following conditions are met:                                  //
+ *                                                                                            //
+ * 1. Redistributions of source code must retain the above copyright notice, this list of     //
+ *    conditions and the following disclaimer.                                                //
+ *                                                                                            //
+ * 2. Redistributions in binary form must reproduce the above copyright notice, this list     //
+ *    of conditions and the following disclaimer in the documentation and/or other            //
+ *    materials provided with the distribution.                                               //
+ *                                                                                            //
+ * 3. Neither the name of the copyright holder nor the names of its contributors may be       //
+ *    used to endorse or promote products derived from this software without specific         //
+ *    prior written permission.                                                               //
+ *                                                                                            //
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY        //
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF    //
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL     //
+ * THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,       //
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,               //
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS    //
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,          //
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF    //
+ * THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.               //
+ */
+
+#ifndef SRC_UTIL_BASE_H_
+#define SRC_UTIL_BASE_H_
+
+#include <fstream>
+#include <iostream>
+#include <string>
+#include <vector>
+#include "util/command.h"
+
+/**
+ * @class BaseCommand
+ * @brief base class for base32 and base64
+ */
+
+class BaseCommand : public Command
+{
+ public:
+  BaseCommand();
+  void PrintUsage(const std::string& cmd_name) const;
+  bool Impl(const std::string&, int argc, const char* argv[]);
+
+ protected:
+  virtual bool
+  do_process(bool encode, std::istream& input, std::ostream& output) = 0;
+  boost::program_options::options_description m_desc;
+  std::string m_opt_type;
+  std::string m_opt_infile;
+  std::string m_opt_outfile;
```

Comment:
I'll comment here as it applies to this whole PR. I'll only comment because you've impressed me so much with your care and attention to detail regarding our style guide, so I know you'll take this message further; and I greatly appreciate that, so thank you @MoroccanMalinois.

An absolute trivial comment: please take note of the upstream style guide (currently Google's style guide) and where it mentions how to label member variables.

This entire codebase is plagued with prepended `m_MixedCase` member variables. It slows down reading in a bad way (and it is generally polluting). I, unfortunately, have kept that tradition since we forked but, because of our recent style change/update, I am looking forward to adhering more to a concrete standard of labeling and I invite you to do so as well. If not, and you think we should adhere to a certain style, please feel free to PR changes to our style guide (I know monero is also big on `m_`, but it somehow works better for them 😄).